mpi4jax/_src/utils.py

```diff
@@ -151,3 +162,8 @@ def has_cuda_support() -> bool:
     from . import xla_bridge
 
     return xla_bridge.HAS_GPU_EXT
+
+
+def prefer_notoken() -> bool:
+    """Returns True if primitive implementations should prefer not to use tokens."""
+    return os.environ.get("MPI4JAX_PREFER_NOTOKEN", "0").lower() in ("1", "true", "on")
```

Comment:
Can we have a way to change the `prefer_notoken` at runtime?

```suggestion

_prefer_notoken = None

def set_prefer_notoken(val: bool):
    global _prefer_notoken
    _prefer_notoken = val
    
def prefer_notoken() -> bool:
    """Returns True if primitive implementations should prefer not to use tokens."""
    global _prefer_notoken
    if _prefer_notoken is None:
         _prefer_notoken = os.environ.get("MPI4JAX_PREFER_NOTOKEN", "0").lower() in ("1", "true", "on")
    return _prefer_notoken 
```

Reply:
Not sure whether that's safe to do. In cases where primitives are already jitted this wouldn't be picked up.

Reply:
yeah, but that would still be safe. It's just that only newly jitted functions will be recompiled.
Anyway, that's fine by me.